trieste/models/interfaces.py

```diff
@@ -152,10 +144,94 @@ def optimize(self, dataset: Dataset) -> None:
         raise NotImplementedError
 
 
-class ModelStack(ProbabilisticModel):
+class SupportsPredictJoint(ProbabilisticModel):
```

Comment:
since most of these are meant to be used by acquisition functions in the BO loop should we make them children of `TrainableProbabilisticModel`?